### PR TITLE
feat(transfers): enable client to check if a quote has expired

### PR DESCRIPTION
**Pull request by RolandSherwin**
## Description

reviewpad:summary

### CODE REVIEW

File: sn_client/src/files.rs [hunk 1]
```diff
@@ -126,7 +126,7 @@ impl FilesApi {
         trace!("Client upload started for chunk: {chunk_addr:?}");
 
         let wallet_client = self.wallet()?;
-        let (payment, payee) = wallet_client.get_recent_payment_for_addr(&chunk_addr)?;
+        let (payment, payee) = wallet_client.get_recent_payment_for_addr(&chunk_addr, true)?;
```
Comment (JasonPaulGithub) [inline sn_client/src/files.rs]: This here is a boolean flag, which is bad design. 
You may not need to apply this right now but it's worth considering this kind of thing in the future.

I have a short article here with reasons and solutions to consider:
https://luzkan.github.io/smells/flag-argument

Comment (JasonPaulGithub) [reply sn_client/src/files.rs]: A more detailed article (if you have time, it's the source article)
https://martinfowler.com/bliki/FlagArgument.html

Comment (RolandSherwin) [reply sn_client/src/files.rs]: Ah yeah, that is a good suggestion. I think we can have multiple functions there instead, as it is a public API.

Comment (RolandSherwin) [reply sn_client/src/files.rs]: Hey @JasonPaulGithub, I've updated the PR!

**Review by joshuef — APPROVED** [review]

**Pull request merged**